sensor/common/networkflow/manager/manager_impl.go

```diff
@@ -506,27 +511,75 @@ func (m *networkFlowManager) enrichAndSendProcesses() {
 	}
 }
 
+func (m *networkFlowManager) handleContainerNotFound(conn *connection, status *connStatus, enrichedConnections map[networkConnIndicator]timestamp.MicroTS) error {
+	var failReasons error
+	failReasons = multierror.Append(failReasons, fmt.Errorf("ContainerID %s unknown", conn.containerID))
+	timeElapsedSinceFirstSeen := timestamp.Now().ElapsedSince(status.firstSeen)
+	// Expire the connection if the container cannot be found within the clusterEntityResolutionWaitPeriod
+	if timeElapsedSinceFirstSeen > maxContainerResolutionWaitPeriod {
+		if activeConn, found := m.activeConnections[*conn]; found {
+			enrichedConnections[*activeConn] = timestamp.Now()
+			delete(m.activeConnections, *conn)
+			flowMetrics.SetActiveFlowsTotalGauge(len(m.activeConnections))
+			return nil // connection has been enriched
+		}
+		status.rotten = true
+		// Only increment metric once the connection is marked rotten
+		flowMetrics.ContainerIDMisses.Inc()
+		log.Debugf("Unable to fetch deployment information for container %s: no deployment found", conn.containerID)
+	} else {
+		failReasons = multierror.Append(failReasons, fmt.Errorf("time for container resolution (%s) not elapsed yet", maxContainerResolutionWaitPeriod))
+	}
+	return failReasons
+}
+
+func formatMultiErrorOneline(errs []error) string {
+	elems := make([]string, len(errs))
+	for i, err := range errs {
+		// The error is used in debug logs and is much nicer to read with the first letter capitalized.
+		// Writing the error message with first capital in the code raises a style warning.
+		msg := cases.Title(language.English, cases.NoLower).String(err.Error())
+		elems[i] = fmt.Sprintf("(%d) %s", i+1, msg)
+	}
+	return strings.Join(elems, ", ")
+}
+
+func logReasonForAggregatingNetGraphFlow(conn *connection, contNs, contName, entitiesName string, port uint16, failReason *multierror.Error) {
+	reasonStr := ""
+	if failReason != nil {
+		failReason.ErrorFormat = formatMultiErrorOneline
+		reasonStr = failReason.Error()
+	}
+	// No need to produce complex chain of reasons, if there is one simple explanation
+	if strings.HasPrefix(conn.remote.IPAndPort.String(), "255.255.255.255") {
```

Comment:
We should probably use the same test as in enrichConnection() (including IPv6):
```
if conn.remote.IPAndPort.Address == externalIPv4Addr || conn.remote.IPAndPort.Address == externalIPv6Addr {
```

Reply:
Well spotted! I addressed that in [fde1828](https://github.com/stackrox/stackrox/pull/13463/commits/fde182895c2727337ec9c6173892673b6a342691)